vignettes/ipums-read.Rmd

```diff
@@ -0,0 +1,390 @@
+---
+title: "Reading IPUMS Data"
+author: "Minnesota Population Center"
+output: rmarkdown::html_vignette
+vignette: >
+  %\VignetteIndexEntry{Reading IPUMS Data}
+  %\VignetteEngine{knitr::rmarkdown}
+  %\VignetteEncoding{UTF-8}
+---
+
+```{r, include = FALSE}
+knitr::opts_chunk$set(
+  collapse = TRUE,
+  comment = "#>"
+)
+```
+
+Once you have downloaded an IPUMS extract, the next step is to load its
+data into R for analysis.
+
+For more information about IPUMS data and how to generate and download a
+data extract, see the [introduction](ipums.html) to IPUMS data.
+
+## IPUMS Extract Structure
+
+IPUMS extracts will be organized slightly differently for different
+[IPUMS projects](https://www.ipums.org/overview). In general, all
+projects will provide multiple files in a data extract. The files most
+relevant to ipumsr are:
+
+-   The metadata file containing information about the variables
+    included in the extract data
+
+-   One or more data files, depending on the project and specifications
+    in the extract
+
+Both of these files are necessary to properly load data into R.
+Obviously, the data files contain the actual data values to be loaded.
+But because these are often in fixed-width format, the metadata files
+are required to correctly parse the data on load.
+
+Even for .csv files, the metadata file allows for the addition of
+contextual variable information to the loaded data. This makes it much
+easier to interpret the values in the data variables and effectively use
+them in your data processing pipeline. See the vignette about [value
+labels](value-labels.html) for more information on working with these
+labels.
+
+## Reading Microdata Extracts
+
+Microdata extracts typically provide their metadata in a DDI (.xml) file
+separate from the compressed data (.dat.gz) files.
+
+Provide the path to the DDI file to `read_ipums_micro()` to directly
+load its associated data file into R.
+
+```{r, message=FALSE}
+library(ipumsr)
+library(dplyr)
+
+# Example data
+cps_ddi_file <- ipums_example("cps_00006.xml")
+
+cps_data <- read_ipums_micro(cps_ddi_file)
+
+head(cps_data)
+```
+
+The loaded data have been parsed correctly and include variable metadata
+in each column. For a summary of the column contents, use
+`ipums_var_info()`:
+
+```{r}
+ipums_var_info(cps_data)
+```
+
+This information is also attached to specific columns. You can obtain it
+with `attributes()` or by using ipumsr helpers:
+
+```{r}
+attributes(cps_data$MONTH)
+
+ipums_val_labels(cps_data$MONTH)
+```
+
+While this is the most straightforward way to load microdata, it's often
```

Comment:
I really like this way of presenting the different workflows for reading in data 👍